File: teambadass/utils/setup_external_storage.py
#!/usr/bin/env python3
"""
TeamBadass External Storage Setup
Creates and configures external storage locations
"""

import os
import sys
import json
import argparse
from pathlib import Path

def main():
    """Main execution function"""
    parser = argparse.ArgumentParser(description="Set up external storage for TeamBadass")
    parser.add_argument('--external-path', type=str, help='Path to external storage location')
    parser.add_argument('--repo-path', type=str, help='Path to TeamBadass repository')
    args = parser.parse_args()
    
    # Set default paths if not provided
    repo_path = args.repo_path or "/home/louthenw/Documents/teambadass_github/teambadass"
    external_path = args.external_path or "/home/louthenw/teambadass_external"
    
    print(f"TeamBadass External Storage Setup")
    print(f"Repository: {repo_path}")
    print(f"External Storage: {external_path}")
    
    # Verify repository exists
    if not os.path.exists(repo_path):
        print(f"Error: Repository path not found: {repo_path}")
        return False
    
    # Create external storage structure
    directories = [
        external_path,
        os.path.join(external_path, "backups"),
        os.path.join(external_path, "temp"),
        os.path.join(external_path, "archive")
    ]
    
    for directory in directories:
        try:
            os.makedirs(directory, exist_ok=True)
            print(f"Created: {directory}")
        except Exception as e:
            print(f"Error creating {directory}: {e}")
            return False
    
    # Create storage configuration
    config = {
        "external_path": external_path,
        "backup_path": os.path.join(external_path, "backups"),
        "temp_path": os.path.join(external_path, "temp"),
        "archive_path": os.path.join(external_path, "archive"),
        "cloud_enabled": False,
        "cloud_provider": None,
        "cloud_config": {}
    }
    
    # Save configuration to repository
    config_path = os.path.join(repo_path, "memory", "config", "storage.json")
    try:
        os.makedirs(os.path.dirname(config_path), exist_ok=True)
        with open(config_path, 'w') as f:
            json.dump(config, f, indent=2)
        print(f"Configuration saved to: {config_path}")
    except Exception as e:
        print(f"Error saving configuration: {e}")
        return False
    
    # Create symbolic link
    try:
        link_path = os.path.join(repo_path, ".external")
        if os.path.exists(link_path) or os.path.islink(link_path):
            os.unlink(link_path)
        
        os.symlink(external_path, link_path)
        print(f"Created symbolic link: {link_path} → {external_path}")
    except Exception as e:
        print(f"Error creating symbolic link: {e}")
        
    print("\nExternal storage setup complete!")
    print("\nTest your setup by running the storage manager:")
    print(f"python3 {os.path.join(repo_path, 'utils', 'storage_manager.py')}")
    
    return True

if __name__ == "__main__":
    sys.exit(0 if main() else 1)

File: teambadass/utils/storage/setup-external-storage.js
#!/usr/bin/env node
/**
 * TeamBadass External Storage Setup
 * Creates and configures external storage locations
 * 
 * FILE_OVERVIEW: Setup utility for external storage configuration
 * VERSION: 1.0.0
 * LAST_UPDATED: 2025-05-03
 * DEPENDENCIES: fs-helper.js
 * 
 * TABLE_OF_CONTENTS:
 * 1. Configuration - Command line argument processing
 * 2. Directory Setup - Creating storage directory structure
 * 3. Configuration Creation - Generating storage configuration
 * 4. Symbolic Link - Creating convenience access links
 * 
 * SKIP_DETAILED_ANALYSIS: True - This header provides sufficient context
 */

const fs = require('fs');
const path = require('path');
const fsHelper = require('../fs-helper');

/**
 * Setup external storage for TeamBadass
 */
async function setupExternalStorage(options = {}) {
  // Default options
  const config = {
    repoPath: options.repoPath || '/home/louthenw/Documents/teambadass_github/teambadass',
    externalPath: options.externalPath || '/home/louthenw/teambadass_external',
    verbose: options.verbose !== false
  };

  console.log('\n📁 TeamBadass External Storage Setup');
  console.log(`📝 Repository: ${config.repoPath}`);
  console.log(`📝 External Storage: ${config.externalPath}`);

  try {
    // Verify repository exists
    const repoExists = await fsHelper.pathExists(config.repoPath);
    if (!repoExists) {
      throw new Error(`Repository path not found: ${config.repoPath}`);
    }

    // Create external storage structure
    const directories = [
      config.externalPath,
      path.join(config.externalPath, 'backups'),
      path.join(config.externalPath, 'temp'),
      path.join(config.externalPath, 'archive')
    ];

    for (const directory of directories) {
      await fsHelper.ensureDirectory(directory);
      console.log(`✅ Created: ${directory}`);
    }

    // Create storage configuration
    const storageConfig = {
      external_path: config.externalPath,
      backup_path: path.join(config.externalPath, 'backups'),
      temp_path: path.join(config.externalPath, 'temp'),
      archive_path: path.join(config.externalPath, 'archive'),
      cloud_enabled: false,
      cloud_provider: null,
      cloud_config: {}
    };

    // Save configuration to repository
    const configPath = path.join(config.repoPath, 'memory', 'config', 'storage.json');
    await fsHelper.ensureDirectory(path.dirname(configPath));
    
    await fsHelper.writeFile(
      configPath,
      JSON.stringify(storageConfig, null, 2)
    );
    
    console.log(`✅ Configuration saved to: ${configPath}`);

    // Create symbolic link
    try {
      const linkPath = path.join(config.repoPath, '.external');
      
      // This would use process.symlink in a real Node.js environment
      console.log(`ℹ️ Would create symbolic link: ${linkPath} → ${config.externalPath}`);
      console.log('   (Symbolic link creation requires Node.js environment)');
    } catch (error) {
      console.log(`⚠️ Error creating symbolic link: ${error.message}`);
    }

    console.log('\n✅ External storage setup complete!');
    console.log('\n📋 Test your setup by using the storage manager:');
    console.log(`   const StorageManager = require('./utils/storage/storage-manager');`);
    console.log(`   const storage = new StorageManager();`);
    console.log(`   storage.initialize().then(result => console.log(result));`);

    return {
      status: 'success',
      config: storageConfig
    };
  } catch (error) {
    console.error(`\n❌ Setup failed: ${error.message}`);
    return {
      status: 'error',
      message: error.message
    };
  }
}

// Execute if run directly
if (require.main === module) {
  // Parse command line arguments
  const args = process.argv.slice(2);
  const options = {};
  
  for (let i = 0; i < args.length; i += 2) {
    const key = args[i].replace(/^--/, '');
    const value = args[i + 1];
    
    if (key && value) {
      options[key] = value;
    }
  }
  
  setupExternalStorage(options)
    .then(result => {
      process.exit(result.status === 'success' ? 0 : 1);
    })
    .catch(error => {
      console.error(`\n❌ Unhandled error: ${error.message}`);
      process.exit(1);
    });
} else {
  // Export for use as a module
  module.exports = setupExternalStorage;
}
